Load testing document (#593)
vdaas/vald: docs/performance/benchmark.md

# Benchmark

> This document explain how to perform a benchmark of approximate nearest neighbor on the Vald cluster, such as [ann-benchmarks](https://github.com/erikbern/ann-benchmarks).

Reviewer: I think this document will show not only ann-benchmarks, cuz code bench sometimes will not relate to ann-benchmark but only performance (transport, load balance, filtering, vectorize, etc...)

> This document explain how to perform a benchmark of approximate nearest neighbor on the Vald cluster, such as [ann-benchmarks](https://github.com/erikbern/ann-benchmarks).

Author: I think we need a separated document to explain the code bench? (line 6) 
That document will explain the unit bench (code bench) for users 
```suggestion
This document will show and explain the benchmark of approximate nearest neighbor on the Vald cluster, such as [ann-benchmarks](https://github.com/erikbern/ann-benchmarks), and also explain how to perform the ann-benchmark test on you Vald cluster.
```

Reviewer: yap.
the main purpose of this doc is what kind of measuring performance in Vald, load test(e2e) and code bench.
And how to use each benchmark tool will be shown in each document, `perfomance/loadtest` and `performance/codebench`.

# Benchmark

> This document will show and explain the benchmark of approximate nearest neighbor on the Vald cluster, such as [ann-benchmarks](https://github.com/erikbern/ann-benchmarks), and also explain how to perform the ann-benchmark test on you Vald cluster.

Reviewer 2: ```suggestion
This document will show and explain the benchmark of an approximate nearest neighbor on the Vald cluster, such as [ann-benchmarks](https://github.com/erikbern/ann-benchmarks), and also explain how to perform the ann-benchmark test on you Vald cluster.
```

If you want to perform load testing on the Vald cluster, please refer to [this document](loadtest.md).
We also created a guideline of the unit benchmark testing, please refer to [this document](unit_benchmark.md).

## Overview

In this document, we will perform the [ann-benchmarks](https://github.com/erikbern/ann-benchmarks) on Vald cluster.

> The test will perform on the whole Vald cluster, which means the request will boardcast to multiple Vald agents instead of single Vald agent, and return the aggreated result.

Reviewer 2: ```suggestion
The test will perform on the whole Vald cluster, which means the request will broadcast to multiple Vald agents instead of single Vald agent, and return the aggreated result.
```

> The result may different from you environment due to the network configuration and overhead of the Vald cluster. In this document we will also explain how to perform the ann-benchmark testing on your environment.

Reviewer 2: ```suggestion
The result may differ from your environment due to the network configuration and overhead of the Vald cluster. In this document, we will also explain how to perform the ann-benchmark testing on your environment.
```

Author: I think it is correct
```suggestion
The result may be different from your environment due to the network configuration and overhead of the Vald cluster. In this document, we will also explain how to perform the ann-benchmark testing on your environment.
```
